src/coreComponents/physicsSolvers/simplePDE/LaplaceVEM.cpp

```diff
@@ -0,0 +1,449 @@
+   - A : "localMatrix" this represents a Compressed Row Storage (optimized for sparse) matrix of real64 values associated with their index,
+   - B : "localRhs" this represents a vector (1d array) of real64 numbers specified at the equation's right-hand side.
+   The "local" prefix indicates that we are working on a local problem here, and the parallelization is performed at a higher level.
+   This assembly step collects all the information needed to create the matrices localMatrix and localRhs, and the computation of values
+   is done in a specific Laplace kernel optimized for parallel performance. Here we:
+   1 - identify and point to the mesh of this domain,
+   2 - find the node manager of this mesh,
+   3 - extract the indices of the nodes that will be solved for (ie. the degrees of freedom or "dof")
+   4 - pass all this information to a Laplace-specific finite element computation kernel.
+   The call to the kernel is a templated call designed for performance (we will not explain the kernel here).
+   See the implementation in LaplaceFEMKernel.cpp.
+ */
+//START_SPHINX_INCLUDE_04
+void LaplaceVEM::assembleSystem( real64 const GEOSX_UNUSED_PARAM( time_n ),
+                                 real64 const GEOSX_UNUSED_PARAM( dt ),
+                                 DomainPartition & domain,
+                                 DofManager const & dofManager,
+                                 CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                 arrayView1d< real64 > const & localRhs )
+{
+  using VEM = virtualElement::ConformingVirtualElementOrder1< m_maxCellNodes, m_maxFaceNodes >;
+
+  MeshLevel & mesh = domain.getMeshBodies().getGroup< MeshBody >( 0 ).getMeshLevel( 0 );
+  ElementRegionManager & elementManager = mesh.getElemManager();
+  NodeManager & nodeManager = mesh.getNodeManager();
+  FaceManager const & faceManager = mesh.getFaceManager();
+  EdgeManager const & edgeManager = mesh.getEdgeManager();
+
+  // Get geometric properties used to compute projectors.
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > nodesCoords =
+    nodeManager.referencePosition();
+  FaceManager::NodeMapType const & faceToNodeMap = faceManager.nodeList();
+  FaceManager::EdgeMapType const & faceToEdgeMap = faceManager.edgeList();
+  EdgeManager::NodeMapType const & edgeToNodeMap = edgeManager.nodeList();
+  arrayView2d< real64 const > const faceCenters = faceManager.faceCenter();
+  arrayView2d< real64 const > const faceNormals = faceManager.faceNormal();
+  arrayView1d< real64 const > const faceAreas = faceManager.faceArea();
+  arrayView1d< globalIndex const > const & dofIndex =
+    nodeManager.getReference< array1d< globalIndex > >( dofManager.getKey( m_fieldName ) );
+
+  real64 const diffusion = 1.0;
+
+  // begin region loop
+  for( localIndex er=0; er<elementManager.numRegions(); ++er )
+  {
+    CellElementRegion const & elementRegion = elementManager.getRegion< CellElementRegion >( er );
```

Comment:
```suggestion
  forTargetRegionsComplete( mesh, [&]( localIndex const,
                                       localIndex const,
                                       ElementRegionBase & elemRegion )
  {

```
to make sure that you only loop over target regions of the solver.